### PR TITLE
CI: Update host OS version, add Python 3.13

### PR DESCRIPTION
**Pull request by EwoutH**
Some CI maintenance:
- Update the Pypi.yml host OSes to their latest compatible versions
- Add Python 3.13 to the test matrix.
  - This needs https://github.com/pyapp-kit/workflows/pull/10

### CODE REVIEW

**Comment by EwoutH** [comment]
If preferred, feel free to work further on this branch!

**Comment by tlambert03** [comment]
thanks, yeah I'm starting to look into it today.  will certainly take some CI adjustment, and it may only be possible to ship the non-mypyc (pure python) wheels at 3.13 release time (mypyc often trails python release dates slightly).  will update!

File: .github/workflows/pypi.yml [hunk 1]
```diff
@@ -56,12 +56,12 @@ jobs:
...
             name: linux-x86_64
-          - os: windows-2019
+          - os: windows-latest
             name: windows-amd64
-          - os: macos-11
+          - os: macos-13
             name: macos-x86_64
             macos_arch: "x86_64"
-          - os: macos-11
+          - os: macos-14
```
Comment (tlambert03) [inline .github/workflows/pypi.yml:64]: @EwoutH, can you comment on these changes?  will these be needed for 3.13?

Comment (EwoutH) [reply .github/workflows/pypi.yml:64]: Not needed an sich, but it's good to compile on the latest host OS versions with the latest toolsets, especially for arm64.

Comment (tlambert03) [reply .github/workflows/pypi.yml:64]: huh, i was previously encouraged to compile platform specific stuff on the oldest compatible toolset...  am I misapplying/misunderstanding old advice?

Comment (EwoutH) [reply .github/workflows/pypi.yml:64]: Generally you use the latest tooling / toolset / toolchain, but with, if needed, a certain target version, which guarantees compatible with that version and up. For example, CMake uses [`MACOSX_DEPLOYMENT_TARGET`](https://cmake.org/cmake/help/latest/envvar/MACOSX_DEPLOYMENT_TARGET.html) for this.

For Python I don't believe it really matters much.

Comment (tlambert03) [reply .github/workflows/pypi.yml:64]: Yeah I remember futzing with macOS deployment target.  Can't remember if it was this project or another, but there was some issue with hatchling not always respecting that env var.  May have been fixed by now.

Anyway, let's just go with it and I'll revisit later if anything fails when it's time to release.  The main python functionality is all working fine on 3.13

Thanks again

**Pull request merged**